### DISCUSSION TITLE
Pickup and Delivery depot node is pickup node?

### DISCUSSION THREAD

**Discussion by vuongdanghuy** [google/or-tools#2735] [Routing (and legacy CP) questions] [answered]
Hi.

I'm new to ortools and I'm solving a pickup and delivery problem. But when i set depot node is one of the pickup nodes, the tool fail with segmentation fault. I don't understand why it's needed for depot node to be different from pickup node.

Could someone explain why?
Thanks.

**Comment by Mizux** [comment]
This is an internal implementation limitation.
You should try to duplicate the depot location by creating a new node and make this one part of the Pickup&Delivery pair...

**Reply by vuongdanghuy** [reply]
Hi Mizux.

Yeah, I create a virtual node to separate depot and pickup node and it works fine. I just don't see why it must be different. If it's internal implementation limitation then ok, i think i have to accept it that way.

**Comment by hklarner** [comment, marked as answer]
I look at it this way: the pickup constraint for the start location
    does not make sense, for two reasons
    
    - if all vehicles start at the depot, it is by definition satisfied
    for all vehicles since it is the first stop of every tour
    - a delivery can only be performed by one vehicle not by all
    vehicles
    
    
    On 23.08.21 04:06, Mio The Best Girl
      wrote:
    
    
      
      Hi.
      I'm new to ortools and I'm solving a pickup and delivery
        problem. But when i set depot node is one of the pickup nodes,
        the tool fail with segmentation fault. I don't understand why
        it's needed for depot node to be different from pickup node.
      Could someone explain why?
        Thanks.
      —
        You are receiving this because you are subscribed to this
        thread.
        Reply to this email directly, view it on GitHub, or unsubscribe.
        Triage notifications on the go with GitHub Mobile for iOS or Android.
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/google/or-tools/discussions/2735",
"url": "https://github.com/google/or-tools/discussions/2735",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by vuongdanghuy** [reply]
Hi hklarner.

Even though it's not true if each vehicle has different start location but I think in general case it's good reason. Thanks.
